library/src/main/java/com/novoda/downloadmanager/NotificationCreator.java

```diff
@@ -1,10 +1,7 @@
 package com.novoda.downloadmanager;
 
-public interface NotificationCreator {
+public interface NotificationCreator<PAYLOAD> {
```

Comment:
We need a `NotificationCreator` for **Migration** and **Downloads**, the client applications should decide how these are displayed. To be able to reuse this interface I have introduced the generic `PAYLOAD` which represents the data need to construct the `Notification` under a specific scenario.

Internally we use `DownloadBatchStatus` for standard downloads, so we type to that. We'll do the same with the migration when we pass that through the builders.

Reply:
This is a good idea, but I'd go with the more usual `T` instead of `PAYLOAD`

Reply:
Yea I figured we wouldn't like this 👅 will change it.

Reply:
85b736f